test/fast_instance2.lean

```diff
@@ -0,0 +1,159 @@
+          (@PartialOrder.mk (Wrapped α)
+            (@Preorder.mk (Wrapped α)
+              (@LE.mk (Wrapped α) fun x y =>
+                @LE.le α
+                  (@Preorder.toLE α
+                    (@PartialOrder.toPreorder α
+                      (@OrderedAddCommGroup.toPartialOrder α
+                        (@StrictOrderedRing.toOrderedAddCommGroup α
+                          (@LinearOrderedRing.toStrictOrderedRing α
+                            (@LinearOrderedCommRing.toLinearOrderedRing α
+                              (@LinearOrderedField.toLinearOrderedCommRing α inst)))))))
+                  (@val α x) (@val α y))
+              (@LT.mk (Wrapped α) fun x y =>
+                @LT.lt α
+                  (@Preorder.toLT α
+                    (@PartialOrder.toPreorder α
+                      (@OrderedAddCommGroup.toPartialOrder α
+                        (@StrictOrderedRing.toOrderedAddCommGroup α
+                          (@LinearOrderedRing.toStrictOrderedRing α
+                            (@LinearOrderedCommRing.toLinearOrderedRing α
+                              (@LinearOrderedField.toLinearOrderedCommRing α inst)))))))
+                  (@val α x) (@val α y))
+              ⋯ ⋯ ⋯)
+            ⋯)
+          ⋯ ⋯ ⋯ ⋯)
+        (@Min.mk (Wrapped α) fun x x_1 => @Inf.inf (Wrapped α) (@instInfWrapped α inst) x x_1)
+        (@Max.mk (Wrapped α) fun x x_1 => @Sup.sup (Wrapped α) (@instSupWrapped α inst) x x_1)
+        (@Ord.mk (Wrapped α) fun a b =>
+          @compare α
+            (@LinearOrder.toOrd α
+              (@LinearOrderedRing.toLinearOrder α
+                (@LinearOrderedCommRing.toLinearOrderedRing α (@LinearOrderedField.toLinearOrderedCommRing α inst))))
+            (@val α a) (@val α b))
+        ⋯
+        (@LinearOrder.decidableLE (Wrapped α)
+          (@LinearOrder.lift (Wrapped α) α
+            (@LinearOrderedRing.toLinearOrder α
+              (@LinearOrderedCommRing.toLinearOrderedRing α (@LinearOrderedField.toLinearOrderedCommRing α inst)))
+            (@instSupWrapped α inst) (@instInfWrapped α inst) (@val α) ⋯ ⋯ ⋯))
+        (@LinearOrder.decidableEq (Wrapped α)
+          (@LinearOrder.lift (Wrapped α) α
+            (@LinearOrderedRing.toLinearOrder α
+              (@LinearOrderedCommRing.toLinearOrderedRing α (@LinearOrderedField.toLinearOrderedCommRing α inst)))
+            (@instSupWrapped α inst) (@instInfWrapped α inst) (@val α) ⋯ ⋯ ⋯))
+        (@LinearOrder.decidableLT (Wrapped α)
+          (@LinearOrder.lift (Wrapped α) α
+            (@LinearOrderedRing.toLinearOrder α
+              (@LinearOrderedCommRing.toLinearOrderedRing α (@LinearOrderedField.toLinearOrderedCommRing α inst)))
+            (@instSupWrapped α inst) (@instInfWrapped α inst) (@val α) ⋯ ⋯ ⋯))
+        ⋯ ⋯ ⋯)
+      ⋯)
+    (@instInvWrapped α inst) (@instDivWrapped α inst) ⋯
+    (@Field.zpow (Wrapped α) (@instField α inst))
```

Comment:
Here we have an argument which is not a class so it is not visited. But it contains a class instance itself as an argument. The current implementation just leaves it as how `Function.Injective` built it. Replacing it wholecloth with the existing instance would avoid having to disassemble the argument during unification.